aidants_connect_web/views/FC_as_FS.py

```diff
@@ -82,13 +81,23 @@ def fc_callback(request):
     }
 
     headers = {"Accept": "application/json"}
+
     request_for_token = python_request.post(token_url, data=payload, headers=headers)
     content = request_for_token.json()
 
     fc_access_token = content.get("access_token")
 
     fc_id_token = content.get("id_token")
-    # TODO Check that ID token is good
+    fc_secret = settings.FC_AS_FS_SECRET
+    fc_id = settings.FC_AS_FS_ID
+    decoded_token = jwt.decode(
+        fc_id_token, fc_secret, audience=fc_id, algorithm="HS256"
+    )
+
+    if connection_state.nonce != decoded_token.get("nonce"):
+        return HttpResponseForbidden()
```

Comment:
It would be interesting to log an error message to we know what triggered the `403`